game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -219,37 +171,40 @@ public static DiceRoll rollDice(
       final String annotation,
       final CombatValue combatValueCalculator) {
 
+    final DiceRoll diceRoll;
+    final PowerStrengthAndRolls unitPowerAndRollsMap =
+        PowerStrengthAndRolls.build(units, combatValueCalculator);
     if (Properties.getLowLuck(bridge.getData().getProperties())) {
-      return rollDiceLowLuck(units, player, bridge, annotation, combatValueCalculator);
+      diceRoll = rollDiceLowLuck(unitPowerAndRollsMap, player, bridge, annotation);
+    } else {
+      diceRoll = rollDiceNormal(unitPowerAndRollsMap, player, bridge, annotation);
     }
-    return rollDiceNormal(units, player, bridge, annotation, combatValueCalculator);
+
+    bridge
+        .getHistoryWriter()
+        .addChildToEvent(annotation + " : " + MyFormatter.asDice(diceRoll), diceRoll);
+    return diceRoll;
   }
 
   /** Roll dice for units using low luck rules. */
   private static DiceRoll rollDiceLowLuck(
-      final Collection<Unit> unitsList,
+      final TotalPowerAndTotalRolls unitPowerAndRollsMap,
       final GamePlayer player,
       final IDelegateBridge bridge,
-      final String annotation,
-      final CombatValue combatValueCalculator) {
-
-    final GameData data = bridge.getData();
+      final String annotation) {
 
-    final int power =
-        PowerStrengthAndRolls.build(unitsList, combatValueCalculator).calculateTotalPower();
+    final int power = unitPowerAndRollsMap.calculateTotalPower();
     if (power == 0) {
       return new DiceRoll(List.of(), 0, 0);
     }
 
     // Roll dice for the fractional part of the dice
-    int hitCount = power / data.getDiceSides();
+    final int diceSides = unitPowerAndRollsMap.getDiceSides();
+    int hitCount = power / diceSides;
```

Comment:
It seems like 'hitCount' should be part of the 'totalPowerAndRolls' interface. Getting data from an object and then using that data to compute a value, as is the case here, appears to be a 'feature-envy' code-smell.

It might be possible that if the API is `getHitCount`, then we won't need the `getDiceSides` method and the dice sides would instead turn into a constructor arg for implementations of `totalPowerAndRolls`

Reply:
In a future PR, I'm going to be reworking this so that it correctly handles units with different dice sides.  At that point, `getDiceSides` should no longer be needed.  I don't think a `getHitCount` method will fit in that rework but I'll look into it when I work on the PR.